libcudacxx/include/cuda/std/__utility/cmp.h

```diff
@@ -31,99 +34,127 @@
 
 _CCCL_PUSH_MACROS
 
+_CCCL_DIAG_PUSH
+_CCCL_DIAG_SUPPRESS_MSVC(4018) // required cast from signed to unsigned
+_CCCL_DIAG_SUPPRESS_MSVC(4388) // required cast from signed to larger unsigned
+_CCCL_DIAG_SUPPRESS_MSVC(4389) // signed/unsigned mismatch for == and !=
```

Comment:
I was thinking about this a bit more, 

We should not get those warnings if we have proper if constexpr support. Can we guard this by
```suggestion
#if defined(_CCCL_NO_IF_CONSTEXPR)
_CCCL_DIAG_SUPPRESS_MSVC(4018) // required cast from signed to unsigned
_CCCL_DIAG_SUPPRESS_MSVC(4388) // required cast from signed to larger unsigned
_CCCL_DIAG_SUPPRESS_MSVC(4389) // signed/unsigned mismatch for == and !=
#endif // _CCCL_NO_IF_CONSTEXPR
```